tools/c7n_azure/c7n_azure/actions.py

```diff
@@ -32,7 +32,29 @@
 from c7n.utils import type_schema
 
 
-class Tag(BaseAction):
+class AzureBaseAction(BaseAction):
+    session = None
+
+    def process(self, resources):
+        self.session = self.manager.get_session()
+        self.process_in_parallel(resources)
+
+    def process_in_parallel(self, resources, max_workers=3, chunk_size=20):
+        return ThreadHelper.execute_in_parallel(
+            resources=resources,
+            execution_method=self.process_resource_set,
+            executor_factory=self.executor_factory,
+            log=self.log,
+            max_workers=max_workers,
+            chunk_size=chunk_size
+        )
+
+    def process_resource_set(self, resources):
```

Comment:
Make this an abstract method `@abstractmethod`

Reply:
Done!